dlib/cmake_utils/dlibConfig.cmake.in

```diff
@@ -28,16 +28,21 @@ if(NOT TARGET dlib-shared AND NOT dlib_BINARY_DIR)
    include("${dlib_CMAKE_DIR}/dlib.cmake")
 endif()
 
-set(dlib_LIBRARIES dlib::dlib)
-set(dlib_LIBS      dlib::dlib)
+find_library(dlib_LIBRARIES dlib HINTS "@CMAKE_INSTALL_FULL_LIBDIR@")
+set(dlib_LIBRARIES ${dlib_LIBRARIES} "@dlib_needed_libraries@")
+set(dlib_LIBS      ${dlib_LIBRARIES} "@dlib_needed_libraries@")
 set(dlib_INCLUDE_DIRS "@CMAKE_INSTALL_FULL_INCLUDEDIR@" "@dlib_needed_includes@")
 
 include(CMakeFindDependencyMacro) 
 
-if("@PNG_FOUND@")
-  find_dependency(PNG CONFIG)
+if(@DLIB_PNG_SUPPORT@)
```

Comment:
@headupinclouds Currently `cmake/Config.cmake.in` used:
* https://github.com/hunter-packages/dlib/blob/bca79d197fbf8ff28798a68e8e0d35a863d71b79/cmake/Config.cmake.in

Reply:
For the header only builds it does, and for the static builds it uses these:

https://github.com/hunter-packages/dlib/blob/bca79d197fbf8ff28798a68e8e0d35a863d71b79/dlib/CMakeLists.txt#L782-L787

Reply:
We can copy the dependency logic from the static lib dlibConfigVersion.cmake to the legacy header only build configuration while it is supported.

Reply:
> For the header only builds it does, and for the static builds it uses these

What a mess :)

I've pushed refactored variant to branch `hunter-19.8-75961ed`:
* https://github.com/hunter-packages/dlib/commits/hunter-19.8-75961ed

Your changes applied as:
* https://github.com/hunter-packages/dlib/commit/bfabe76fdce5eedb5dff7399a14cc2f6d8e908ea
* https://github.com/hunter-packages/dlib/commit/c7489fdd15c020440c6a8ae5511639dd2d63b28a
* https://github.com/hunter-packages/dlib/commit/af4a88734e5196cc2da64dc00c72351251fba27d
* https://github.com/hunter-packages/dlib/commit/7512623e261b0a21bc5ec297019857aff09efb38